roster/model.py

```diff
@@ -0,0 +1,137 @@
+"""Internal roster representation."""
+
+from __future__ import annotations
+
+__LICENSE__ = """
+Copyright 2019 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import re
+
+from enum import Enum
+from typing import Set, List, Dict, Optional
+
+
+class CharacterRole(Enum):
+    """List of possible roles in the game."""
+    TANK = "TANK"
+    HEALER = "HEALER"
+    DPS = "DPS"
+
+
+class CharacterClass(Enum):
+    """List of supported classes in the game."""
+    DEATH_KNIGHT = "Death Knight"
+    DEMON_HUNTER = "Demon Hunter"
+    DRUID = "Druid"
+    HUNTER = "Hunter"
+    MAGE = "Mage"
+    MONK = "Monk"
+    PALADIN = "Paladin"
+    PRIEST = "Priest"
+    ROGUE = "Rogue"
+    SHAMAN = "Shaman"
+    WARLOCK = "Warlock"
+    WARRIOR = "Warrior"
+
+    @staticmethod
+    def get_all_roles(klass: 'CharacterClass') -> List[CharacterRole]:
```

Comment:
Because it's not defined yet at runtime. Long story short, as long as the indentation is not fully reverted to the `class` level, the name of the class is undefined yet. However if a `def` statement or sub `class` statement occurs, this will be lazy evaluated parsed slightly later, thus the name of the class may exist at this point (relevant to the @staticmethod statements).